rust/monovertex/src/transformer.rs

```diff
@@ -1,61 +1,160 @@
-use crate::error::Result;
-use crate::message::Message;
+use std::collections::HashMap;
+
+use crate::config::config;
+use crate::error::{Error, Result};
+use crate::message::{Message, Offset};
 use crate::shared::utc_from_timestamp;
-use crate::sourcetransform_pb::source_transform_client::SourceTransformClient;
-use crate::sourcetransform_pb::SourceTransformRequest;
+use crate::sourcetransform_pb::{
+    self, source_transform_client::SourceTransformClient, SourceTransformRequest,
+    SourceTransformResponse,
+};
+
+use tokio::sync::mpsc;
+use tokio::task::JoinHandle;
+use tokio::time::Duration;
+use tokio_stream::wrappers::ReceiverStream;
+use tokio_util::sync::CancellationToken;
 use tonic::transport::Channel;
+use tonic::{Request, Streaming};
 
 const DROP: &str = "U+005C__DROP__";
 
 /// TransformerClient is a client to interact with the transformer server.
-#[derive(Clone)]
+#[derive(Debug)]
 pub struct SourceTransformer {
-    client: SourceTransformClient<Channel>,
+    read_tx: mpsc::Sender<SourceTransformRequest>,
+    resp_stream: Streaming<SourceTransformResponse>,
 }
 
 impl SourceTransformer {
-    pub(crate) async fn new(client: SourceTransformClient<Channel>) -> Result<Self> {
-        Ok(Self { client })
+    pub(crate) async fn new(mut client: SourceTransformClient<Channel>) -> Result<Self> {
+        let (read_tx, read_rx) = mpsc::channel(config().batch_size as usize);
```

Comment:
NIT: referencing `config` directly can make testing difficult if you want to change the value  for testing.